src/components/pages/swap/__test__/Swap.test.jsx

```diff
@@ -81,6 +84,106 @@ describe('Swap', async () => {
     expect(swapButton).toBeDisabled()
   })
 
+  it('Should prevent USDT swap from Algorand if pool is not reachable', async () => {
+    vi.spyOn(UseSwapInfo, 'useSwapInfo').mockReturnValue({ eta: '1', poolAmount: undefined })
+    vi.spyOn(SwapInfo, 'default').mockImplementation(() => <div data-testid="swap-info" />)
+    vi.spyOn(feeUtils, 'getSwapFees').mockResolvedValue({ basisPoints: 15, networkFee: 1e18, minProtocolFee: 2e18 })
+    render(<Wrapper asset="usdt" originBlockchain="algorand" destBlockchain="eth" algorand_from_assetid={'312769'} />)
+    await waitFor(() => expect(screen.getByText(/Unknown liquidity/)).toBeInTheDocument())
+    const [swapButton] = screen.getAllByRole('button')
+    const [fromInput, toInput] = screen.getAllByRole('textbox')
+    await UserEvent.type(fromInput, '1')
+    expect(fromInput).toHaveAttribute('value', '1')
+    expect(toInput).toHaveAttribute('value', '-2')
+    expect(swapButton).toHaveTextContent('Unknown liquidity')
+    expect(screen.getByText(/It is currently not possible to check the/)).toBeInTheDocument()
+    expect(swapButton).toBeDisabled()
+  })
+
+  it('Should continue and warn USDT swap from Algorand if pool liquidity is insufficient', async () => {
```

Comment:
```suggestion
  it('Should prevent...', async () => {
```
you are asserting the button is disabled here